src/main/java/com/github/mauricioaniche/ck/metric/NumberOfFields.java

```diff
@@ -18,7 +24,7 @@
 	@Override
 	public void visit(FieldDeclaration node) {
 		fields++;
-
+		fieldNames.add(getVariableName(node.fragments()));
```

Comment:
The list of fragments could be typed, so that you would let JDT to parse the name, and you don't have to think about the name different ways to declare a field and propose regexes.

According to the [documentation](https://www.ibm.com/support/knowledgecenter/SSRTLW_9.5.0/org.eclipse.jdt.doc.isv/reference/api/org/eclipse/jdt/core/dom/FieldDeclaration.html#fragments--), method `fragments()`:

_Returns the live list of variable declaration fragments in this field declaration. Adding and removing nodes from this list affects this node dynamically. All nodes in this list must be VariableDeclarationFragments; attempts to add any other type of node will trigger an exception._

Reply:
Thanks for the hint. :)